src/vsr/replica.zig

```diff
@@ -1484,8 +1449,17 @@ pub fn Replica(
             // The list of remote replicas yet to send a prepare_ok:
             var waiting: [config.replicas_max]u8 = undefined;
             var waiting_len: usize = 0;
-            for (prepare.ok_from_all_replicas[0..self.replica_count]) |received, replica| {
-                if (received == null and replica != self.replica) {
+            var ok_from_all_replicas_iterator = prepare.ok_from_all_replicas.iterator(.{
+                .kind = .unset,
+            });
+            while (ok_from_all_replicas_iterator.next()) |replica| {
+                // Ensure we don't wait for replicas that don't exist.
+                if (replica == self.replica_count) {
```

Comment:
Are we guaranteed to hit this exit condition always?

The comment is good. It would be good to further explain a little more why the captured `replica` might go beyond `replica_count`.

Reply:
> Are we guaranteed to hit this exit condition always?

No, only when `self.replica_count < config.replicas_max` (since the bitset has `config.replicas_max` bits, and we are iterating over _unset_ bits, but only want to count the ones below `self.replica_count`). The assert in this block hinted at it, but I'll add more of an explanation in the comment.

Reply:
https://github.com/coilhq/tigerbeetle/pull/72/commits/b40cc73ecb2c31ef9d9e9ea6ffcdd91675054c8b